test/csuite/wt5212_incremental_backup/main.c

```diff
@@ -0,0 +1,779 @@
+            if (strncmp(filename, WTLOG, WTLOGLEN) == 0)
+                testutil_check(__wt_snprintf(buf, sizeof(buf), "cp %s/%s/%s %s/%s/%s", home,
+                  logpath, filename, h, logpath, filename));
+            else
+                testutil_check(
+                  __wt_snprintf(buf, sizeof(buf), "cp %s/%s %s/%s", home, filename, h, filename));
+            testutil_check(system(buf));
+        }
+    }
+    scan_end_check(ret == WT_NOTFOUND);
+
+    /* Done processing all files. Close backup cursor. */
+    testutil_check(backup_cur->close(backup_cur));
+    finalize_files(flist, count);
+    free(tmp);
+    /*! [incremental backup using block transfer]*/
+}
+
+static void
+add_data_validate_backups(WT_SESSION *session)
+{
+    int i, j;
+    printf("Adding initial data\n");
+    add_work(session, 0, 0, uri, uri2);
+
+    printf("Taking initial backup\n");
+    take_full_backup(session, 0);
+
+    testutil_check(session->checkpoint(session, NULL));
+
+    for (i = 1; i < ITERATIONS_MULTIPLIER; i++) {
+        printf("Iteration %d: adding data\n", i);
+        /* For each iteration we may add work and checkpoint multiple times. */
+        for (j = 0; j < i; j++) {
+            add_work(session, i, j, uri, uri2);
+            testutil_check(session->checkpoint(session, NULL));
+        }
+
+        /*
+         * The full backup here is only needed for testing and comparison purposes. A normal
+         * incremental backup procedure would not include this.
+         */
+        printf("Iteration %d: taking full backup\n", i);
+        take_full_backup(session, i);
+        /*
+         * Taking the incremental backup also calls truncate to archive the log files, if the copies
+         * were successful. See that function for details on that call.
+         */
+        printf("Iteration %d: taking incremental backup\n", i);
+        take_incr_backup(session, i);
+
+        printf("Iteration %d: dumping and comparing data\n", i);
+        testutil_check(compare_backups(i, uri));
+    }
+}
+
+/*
+ * Remove all the records, take backups and validate
+ */
+static void
+remove_all_records_validate(WT_SESSION *session)
+{
+    int i, j;
+
+    remove_work(session, 0, 0);
+    for (i = 1; i < ITERATIONS_MULTIPLIER; i++) {
+        printf("Iteration %d: Removing data\n", i);
+        for (j = 0; j < i; j++) {
+            remove_work(session, i, j);
+        }
+    }
+    take_full_backup(session, i);
+
+    take_incr_backup(session, i);
+
+    printf("Dumping and comparing data\n");
+    testutil_check(compare_backups(i, uri));
+}
+
+static void
+drop_old_add_new_objects(WT_CONNECTION *wt_conn)
+{
+    WT_CURSOR *backup_cur;
+    WT_SESSION *session;
+    int ret, i, j;
+    char buf[1024];
+
+    testutil_check(wt_conn->close(wt_conn, NULL));
+    testutil_check(wiredtiger_open(home, NULL, CONN_CONFIG, &wt_conn));
+    testutil_check(wt_conn->open_session(wt_conn, NULL, NULL, &session));
```

Comment:
Why does this function, and the next function, both close and reopen the connection and session? Does it need to do that? Does that add something to the test?

Reply:
It throws `"Device or resource busy"` error if I try to drop an object without doing this. So, I have updated the code to do this just before dropping a table i.e, before `drop_old_add_new_table` function and removed from other places.